doc/extract_swagger.py

```diff
@@ -18,8 +18,8 @@ async def extract_swagger(destination_fn):
     config.api.http_enabled = False
     config.api.https_enabled = False
 
-    root_endpoint = RootEndpoint(config, middlewares=[ApiKeyMiddleware(config.api.key), error_middleware])
-    api_manager = RESTManager(config=config.api, root_endpoint=root_endpoint, state_dir=config.state_dir)
+    root_endpoint = RootEndpoint(middlewares=[ApiKeyMiddleware(config.api.key), error_middleware])
+    api_manager = RESTManager(config.api, root_endpoint, config.state_dir)
```

Comment:
**NIT** seems that this change is unnecessary.

Reply:
(Removing `config` argument from `RootEndpoint` is necessary.)
The change from named to unnamed arguments is to conform to the way of indicating required vs optional arguments:
required args must be unnamed, while optional must be named.

Reply:
Is it a common practice? If yes, could you post a link to the resource that profs that?

Reply:
![изображение](https://user-images.githubusercontent.com/2509103/143269681-85603578-9066-4208-90f0-6d78a8d4716e.png)


https://www.python.org/dev/peps/pep-0570/#benefits-of-positional-only-parameters


An alternative opinion:
https://trstringer.com/python-named-arguments/

Reply:
The screenshot above (and link to the python.org) doesn't refer to 

>The change from named to unnamed arguments is to conform to the way of indicating required vs optional arguments: required args must be unnamed, while optional must be named.

Reply:
Let's put it a different way: in some cases (e.g. C-based functions), using named parameters can lead to incompatibility.
For instance, let's say one day we decide to replace some class definitions with C-based counterparts. Then, named arguments won't work anymore. CPython makes the false promise of allowing addressing positional arguments by name, as shown in the screenshot. Therefore, we must adhere to the safest option possible: address positional arguments by position, and addressing named arguments by name. Another argument goes for subclassing and avoiding Liskov violation:
![изображение](https://user-images.githubusercontent.com/2509103/143278073-fa17e1e6-7a11-4bb3-9501-15f11477f102.png)

Yet another argument is the one of _consistency_: this is the first time I see someone address positional arguments by name in our codebase. If we decide to allow addressing positional arguments by name, we must provide guidelines for that (e.g. "always address by name"). Otherwise, this is surprising and inconsistent.

Reply:
> Let's put it a different way: in some cases (e.g. C-based functions), using named parameters can lead to incompatibility.

Agree, but there are no C-based functions here. We should not treat all of our functions as C-basedso because of the potential presence of C-based functions somewhere.

> Yet another argument is the one of consistency: this is the first time I see someone address positional arguments by name in our codebase.

This change was introduced by @xoriole. I also use this approach during writing code for Tribler because it increases readability.

> Another argument goes for subclassing and avoiding Liskov violation:
I don't see any application of Liskov violation in our case. 
Even if you want to refactor some parts of Tribler by replacing them with C-based counterparts there are plenty of ways how to do it without Liskov violation. 
It is also an extremely unlikely scenario that we would change `RESTManager` to a C-based counterpart.

Reply:
For the sake of keeping the team productive, and not discussing petty style issues, I'm reverting this change. In the future, I will not do this kind of change to prevent such useless disputes. :shrug: